src/SFA.DAS.EmployerCommitmentsV2.Web/Controllers/ApprenticeController.cs

```diff
@@ -1,45 +1,65 @@
-using SFA.DAS.Employer.Shared.UI;
-using SFA.DAS.Employer.Shared.UI.Attributes;
-using SFA.DAS.EmployerCommitmentsV2.Features;
-using SFA.DAS.EmployerCommitmentsV2.Web.Models.Apprentice;
-using SFA.DAS.EmployerCommitmentsV2.Web.RouteValues;
-
-namespace SFA.DAS.EmployerCommitmentsV2.Web.Controllers
-{
-    [Route("{accountHashedId}/apprentices")]
-    [SetNavigationSection(NavigationSection.ApprenticesHome)]
-    [DasAuthorize(EmployerUserRole.OwnerOrTransactor)]
-    public class ApprenticeController : Controller
-    {
-        private readonly IModelMapper _modelMapper;
-
-        public ApprenticeController(IModelMapper modelMapper)
-        {
-            _modelMapper = modelMapper;
-        }
-
-        [Route("", Name = RouteNames.ApprenticesIndex)]
-        [DasAuthorize(EmployerFeature.ManageApprenticesV2)]
-        public async Task<IActionResult> Index(IndexRequest request)
-        {
-            var viewModel = await _modelMapper.Map<IndexViewModel>(request);
-            viewModel.SortedByHeader();
-
-            return View(viewModel);
-        }
-
-        [Route("download", Name = RouteNames.ApprenticesDownload)]
-        [DasAuthorize(EmployerFeature.ManageApprenticesV2)]
-        public async Task<IActionResult> Download(DownloadRequest request)
-        {
-            var downloadViewModel = await _modelMapper.Map<DownloadViewModel>(request);
-
-            return File(downloadViewModel.Content, Constants.ApprenticesSearch.DownloadContentType, downloadViewModel.Name);
-        }
-    }
+﻿using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using SFA.DAS.Authorization.EmployerUserRoles.Options;
+using SFA.DAS.Authorization.Mvc.Attributes;
+using SFA.DAS.CommitmentsV2.Shared.Interfaces;
+using SFA.DAS.Employer.Shared.UI;
+using SFA.DAS.Employer.Shared.UI.Attributes;
+using SFA.DAS.EmployerCommitmentsV2.Features;
+using SFA.DAS.EmployerCommitmentsV2.Web.Cookies;
+using SFA.DAS.EmployerCommitmentsV2.Web.Models.Apprentice;
+using SFA.DAS.EmployerCommitmentsV2.Web.RouteValues;
+
+namespace SFA.DAS.EmployerCommitmentsV2.Web.Controllers
+{
+    [Route("{accountHashedId}/apprentices")]
+    [SetNavigationSection(NavigationSection.ApprenticesHome)]
+    [DasAuthorize(EmployerUserRole.OwnerOrTransactor)]
+    public class ApprenticeController : Controller
+    {
+        private readonly IModelMapper _modelMapper;
+        private readonly ICookieStorageService<IndexRequest> _cookieStorage;
+
+        public ApprenticeController(IModelMapper modelMapper, ICookieStorageService<IndexRequest> cookieStorage)
+        {
+            _modelMapper = modelMapper;
+            _cookieStorage = cookieStorage;
+        }
+
+        [Route("", Name = RouteNames.ApprenticesIndex)]
+        [DasAuthorize(EmployerFeature.ManageApprenticesV2)]
+        public async Task<IActionResult> Index(IndexRequest request)
+        {
+            IndexRequest savedRequest = null;
+
+            if (request.FromSearch)

```

Comment:
I would have thought that this would be nullable

Reply:
No need as we just need to check for whether its true.